geozero/src/geojson/geojson_line_writer.rs

```diff
@@ -0,0 +1,190 @@
+        m: Option<f64>,
+        t: Option<f64>,
+        tm: Option<u64>,
+        idx: usize,
+    ) -> Result<()> {
+        self.line_writer.coordinate(x, y, z, m, t, tm, idx)
+    }
+
+    fn empty_point(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.empty_point(idx)
+    }
+
+    fn point_begin(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.point_begin(idx)
+    }
+
+    fn point_end(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.point_end(idx)
+    }
+
+    fn multipoint_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.multipoint_begin(size, idx)
+    }
+
+    fn multipoint_end(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.multipoint_end(idx)
+    }
+
+    fn linestring_begin(&mut self, tagged: bool, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.linestring_begin(tagged, size, idx)
+    }
+
+    fn linestring_end(&mut self, tagged: bool, idx: usize) -> Result<()> {
+        self.line_writer.linestring_end(tagged, idx)
+    }
+
+    fn multilinestring_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.multilinestring_begin(size, idx)
+    }
+
+    fn multilinestring_end(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.multilinestring_end(idx)
+    }
+
+    fn polygon_begin(&mut self, tagged: bool, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.polygon_begin(tagged, size, idx)
+    }
+
+    fn polygon_end(&mut self, tagged: bool, idx: usize) -> Result<()> {
+        self.line_writer.polygon_end(tagged, idx)
+    }
+
+    fn multipolygon_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.multipolygon_begin(size, idx)
+    }
+
+    fn multipolygon_end(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.multipolygon_end(idx)
+    }
+
+    fn geometrycollection_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.line_writer.geometrycollection_begin(size, idx)
+    }
+
+    fn geometrycollection_end(&mut self, idx: usize) -> Result<()> {
+        self.line_writer.geometrycollection_end(idx)
+    }
+}
+
+impl<W: Write> PropertyProcessor for GeoJsonLineWriter<W> {
+    fn property(&mut self, idx: usize, name: &str, value: &crate::ColumnValue) -> Result<bool> {
+        self.line_writer.property(idx, name, value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::geojson::read_geojson_lines;
+
+    #[ignore = "fails because no newlines are written. feature_end is never called."]
+    #[test]
+    fn good_geometries() {
+        let input = r#"{ "type": "Point", "coordinates": [1.1, 1.2] }
+    { "type": "Point", "coordinates": [2.1, 2.2] }
+    { "type": "Point", "coordinates": [3.1, 3.2] }
+    "#;
+        let mut out: Vec<u8> = Vec::new();
+        assert!(
+            read_geojson_lines(input.as_bytes(), &mut GeoJsonLineWriter::new(&mut out)).is_ok()
```

Comment:
Maybe we could count how many `*_begin` are called, and then we could print a newline whenever the corresponding number of `*_end` are called.

As long as the data and processor is valid, that should work. Though maybe #184 precludes that from working with points. If we can't catch this with the compiler, maybe a runtime error is better than quietly producing invalid output?

Reply:
Oh I forgot #184 only affects multipoint... I think something like this might work:
```
GeoJsonLineWriter {
   open_contexts: 0
   ...
}

impl GeoJsonLineWriter {
    fn begin_context(&mut self) {
        self.open_contexts +=1;
    }
    fn end_context(&mut self) {
        self.open_context -=1;
        if self.open_contexts == 0 {
            self.write_newline();
        }
    }
} 

impl FeatureProcessor for GeoJsonLineWriter {
    fn feature_begin(&mut self) {
       self.begin_context()
    }
    fn feature_end(&mut self) {
       self.end_context()
    }
}
impl GeomProcessor for GeoJsonLineWriter {
    fn point_begin(&mut self) {
        self.begin_context()
        ...
    }
    fn point_end(&mut self) {
        self.end_context()
        ...
    }
    fn linestring_begin(&mut self) {
        self.begin_context()
        ...
    }
    fn linestring_end(&mut self) {
        self.end_context()
        ...
    }

}
```

Reply:
That makes sense, thanks for the idea!